product_subscription/models/product_release.py

```diff
@@ -115,89 +116,86 @@ def action_view_delivery(self):
             result['res_id'] = pick_ids[0]
         return result
 
-    @api.one
+    @api.multi
     def action_draft(self):
+        self.ensure_one()
         self.product_release_lines.unlink()
         self.state = 'draft'
 
-    @api.one
+    @api.multi
     def action_cancel(self):
+        self.ensure_one()
         self.product_release_lines.unlink()
         if self.state in ['draft', 'validated']:
             self.state = 'cancelled'
 
-    @api.one
+    @api.multi
     def action_validate(self):
-        prod_sub_obj = self.env['product.subscription.object']
-        release_line_obj = self.env['product.release.line']
+        self.ensure_one()
 
-        vals = {'state': 'validated'}
         if self.name == '' or not self.name:
             prod_rel_seq = self.env.ref('product_subscription.sequence_product_release', False)  # noqa
-            vals['name'] = prod_rel_seq.next_by_id()
-
-        subscriptions = prod_sub_obj.search([('counter', '>', 0)])
+            name = prod_rel_seq.next_by_id()
+        else:
+            name = self.name
 
```

Comment:
why setting name to its name ? I prefer doing it with vals

Reply:
That's what I do at line 156. I need to set the `name` variable in the else to make sure it's in the scope.

Reply:
This is why vals was for at the first place. you only set name in vals dict if needed and you avoid an useless else statement. As state will always be 'validated' setting name is just for the case where it has no name already.